### ISSUE TITLE
Unexpected FlowExpression: FlowExpressions$Unknown

### ISSUE THREAD

**Issue by vlsi** [typetools/checker-framework#3631]
File in question: https://github.com/vlsi/calcite/blob/908c35d9ae2188c79e37df5d085f36f9d989a797/core/src/main/java/org/apache/calcite/rex/RexCall.java#L130

The stacktrace is as follows:

```
  ; The Checker Framework crashed.  Please report the crash.
  Compilation unit: core/src/main/java/org/apache/calcite/rex/RexCall.java
  Last visited tree at line 130 column 63:
        operandDigests.add(((RexLiteral) operand).computeDigest(includeType));
  Exception: java.lang.Throwable; java.lang.Throwable
        at org.checkerframework.javacutil.BugInCF.<init>(BugInCF.java:14)
        at org.checkerframework.framework.flow.CFAbstractStore.getValue(CFAbstractStore.java:525)
        at org.checkerframework.checker.initialization.InitializationVisitor.checkContract(InitializationVisitor.java:200)
        at org.checkerframework.common.basetype.BaseTypeVisitor.checkPreconditions(BaseTypeVisitor.java:1589)
        at org.checkerframework.common.basetype.BaseTypeVisitor.visitMethodInvocation(BaseTypeVisitor.java:1401)
        at org.checkerframework.checker.nullness.NullnessVisitor.visitMethodInvocation(NullnessVisitor.java:482)
        at org.checkerframework.checker.nullness.NullnessVisitor.visitMethodInvocation(NullnessVisitor.java:59)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1650)
        at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
        at org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:88)
        at org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:290)
        at org.checkerframework.common.basetype.BaseTypeVisitor.checkArguments(BaseTypeVisitor.java:2995)
        at org.checkerframework.common.basetype.BaseTypeVisitor.visitMethodInvocation(BaseTypeVisitor.java:1386)
        at org.checkerframework.checker.nullness.NullnessVisitor.visitMethodInvocation(NullnessVisitor.java:482)
        at org.checkerframework.checker.nullness.NullnessVisitor.visitMethodInvocation(NullnessVisitor.java:59)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1650)
        at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
        at org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:88)
        at org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:290)
        at org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:167)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitExpressionStatement(TreeScanner.java:433)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCExpressionStatement.accept(JCTree.java:1460)
        at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
        at org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:88)
        at org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:290)
        at org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:167)
        at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
```

**Comment by mernst** [comment]
Thanks for the bug report.  I'm sorry you are having trouble.  

Can you please tell us how to reproduce the problem?  What is a command I can run?

I tried compiling the file in isolation, but that resulted in no crash.

I was not able to find 908c35d9ae2188c79e37df5d085f36f9d989a797 or 0e13d65e4736457a776d3734c4a9b99ff81a719e in https://github.com/vlsi/calcite .

**Comment by vlsi** [comment]
Please try `checkerframework` branch.

The command is `./gradlew classes -PenableCheckerframework`.
The error is slightly different now, so it is not clear if the location is really relevant to the error itself. I tried to tweak the code around, however, small changes like extracting variables produce no effect.

```
  Last visited tree at line 124 column 16:
              || ((RexLiteral) otherArg).digestIncludesType() == RexDigestIncludeType.NO_TYPE)
  Exception: java.lang.Throwable; java.lang.Throwable
```

**Comment by mernst** [comment]
That enables me to reproduce the crash.  Thanks!

**Comment by mernst** [comment]
Here is a smaller test case (run it with the Nullness Checker):

```java
import org.checkerframework.checker.nullness.qual.RequiresNonNull;

public class Issue3631 {

    void f(Object otherArg) {
        ((Issue3631Helper) otherArg).m();
    }
}

class Issue3631Helper {

    String type = "foo";

    @RequiresNonNull("type")
    void m() {}
}
```

**Comment by vlsi** [comment]
This crash does not appear in 3.6.0, and it appears in 3.6.1

**Comment by vlsi** [comment]
The workaround is to add a wrapper method as follows, then `workaround3631((Issue3631Helper) otherArg)` works 

```java
class Issue3631Helper {

    String type = "foo";

    @RequiresNonNull("type")
    void m() {}

    void workaround3631(Issue3631Helper x) {
         x.m();
    }
}
```

**Comment by mernst** [comment]
This was fixed in 3364fdfee979ad2ba1200855a843eac9f5086d43.
